WordPress/src/main/java/org/wordpress/android/ui/stats/StatsActivity.java

```diff
@@ -180,6 +180,14 @@ public void run() {
             }
         }
 
+        if (mSite == null) {
+            ToastUtils.showToast(this, R.string.blog_not_found, ToastUtils.Duration.SHORT);
+            finish();
+            return;
+        }
+
+        // If the site is not connected via wpcom (Jetpack included), then show a dialog to the user.
+        checkIfSiteHasAccessibleStats(mSite);
```

Comment:
Somewhere around here we need to make sure the user is actually signed into WP.com.

From a clean app, add a self-hosted site that is Jetpack-connected, and visit stats. You get the old repeated message shown in https://github.com/wordpress-mobile/WordPress-Android/issues/5032, because we determined that it's WP API-accessible but haven't verified a token.